src/charm.py

```diff
@@ -219,27 +235,74 @@ def self_scraping_job(self):
         """The self-monitoring scrape job."""
         external_url = urlparse(self._external_url)
         metrics_path = f"{external_url.path.rstrip('/')}/metrics"
-        target = (
-            f"{external_url.hostname}{':'+str(external_url.port) if external_url.port else ''}"
-        )
+        internal_url = self._internal_url.replace("http://", "")
+        target = f"{internal_url}"
         job = {
             "metrics_path": metrics_path,
             "static_configs": [{"targets": [target]}],
         }
 
         return [job]
 
+    def _update_blackbox_config_yaml_from_relation(self, modules) -> None:
+        """Update the blackbox config yaml with modules defined in relation.
```

Comment:
This method doesn't work with relations directly. Should we say "given dict" instead of "relation"?